app/init.php

```diff
@@ -1246,6 +1250,38 @@ function (mixed $value) {
     return $project;
 }, ['dbForConsole', 'request', 'console']);
 
+App::setResource('clientSession', function (string $mode, Document $project, Document $console, Appwrite\Utopia\Request $request) {
+    Auth::setCookieName('a_session_' . $project->getId());
+
+    if (APP_MODE_ADMIN === $mode) {
+        Auth::setCookieName('a_session_' . $console->getId());
+    }
+
+    $session = Auth::decodeSession(
+        $request->getCookie(
+            Auth::$cookieName, // Get sessions
+            $request->getCookie(Auth::$cookieName . '_legacy', '')
+        )
+    );
+
+    // Get session from header for SSR clients
+    if (empty($session['id']) && empty($session['secret'])) {
+        $sessionHeader = $request->getHeader('x-appwrite-session', '');
+
+        if (!empty($sessionHeader)) {
+            $session = Auth::decodeSession($sessionHeader);
+        }
+    }
+
+    if (empty($session['id']) && empty($session['secret'])) {
+        $fallback = $request->getHeader('x-fallback-cookies', '');
+        $fallback = \json_decode($fallback, true);
+        $session = Auth::decodeSession(((isset($fallback[Auth::$cookieName])) ? $fallback[Auth::$cookieName] : ''));
+    }
+
+    return $session;
+}, ['mode', 'project', 'console', 'request']);
+
```

Comment:
Can we avoid setting an additional resource considering we already have the `user` resource?

Also, we have a `session` resource below this.